src/main/java/org/opensearch/sdk/ExtensionsRunner.java

```diff
@@ -167,8 +174,12 @@ protected ExtensionsRunner(Extension extension) throws IOException {
         this.customSettings = extension.getSettings();
         // save custom namedXContent
         this.customNamedXContent = extension.getNamedXContent();
+        // save custom namedWriteable
+        this.customNamedWriteable = extension.getNamedWriteable();
         // initialize NamedXContent Registry. Must happen after getting extension namedXContent
         this.sdkNamedXContentRegistry = new SDKNamedXContentRegistry(this);
+        // initialize NamedWriteable Registry. Must happen after getting extension namedWriteable
+        this.sdkNamedWriteableRegistry = new SDKNamedWriteableRegistry(this);
```

Comment:
Let's add this variable (via its getter) to the injected components in Guice (below line 196).